### PR TITLE
feat: add process request seconds for executor

### PR DESCRIPTION
**Pull request by samsja**
Add `process_request_seconds` (on requests method) as a prometheus metrics to the Executor.

The goal of this metric is to monitor how much time each method decorated with the `@requests` takes to be processed

The metrics name is the same for all of the decorated method but there is a label for filtering.

Screenshot from 2022-04-19 15-29-08

### CODE REVIEW

File: jina/serve/executors/decorators.py [hunk 1]
```diff
@@ -129,6 +147,15 @@ def __set_name__(self, owner, name):
 
             setattr(owner, name, self.fn)
 
+        def _get_summary(
+            self, summary: Optional['Summary'], method_name: str
+        ) -> ContextManager:
+            return (
+                summary.labels(method_name).time()
```
Comment (jacobowitz) [inline jina/serve/executors/decorators.py:154]: Can we also add the executor name as a label?

File: jina/serve/executors/decorators.py [hunk 2]
```diff
@@ -104,15 +115,22 @@ def __init__(self, fn):
...
+                        return await fn(self2, *args, **kwargs)
 
                 self.fn = arg_wrapper
             else:
 
                 @functools.wraps(fn)
-                def arg_wrapper(*args, **kwargs):
-                    return fn(*args, **kwargs)
+                def arg_wrapper(
+                    self2, *args, **kwargs
```
Comment (jacobowitz) [inline jina/serve/executors/decorators.py:129]: maybe lets rename self2 to sth else? Is this conceptually a `self`? Its just `executor_instance` or sth no??

**Review by jacobowitz — CHANGES_REQUESTED** [review]
this is pretty cool!

File: jina/serve/executors/decorators.py (hunk 1 above)
Comment (samsja) [reply jina/serve/executors/decorators.py:154]: Great idea, I will add it

**Review by jacobowitz — APPROVED** [review]
this will be very helpful!

**Review by jacobowitz — APPROVED** [review]

**Pull request merged**